### DISCUSSION TITLE
Next.js with Next-auth, wrong VERCEL_URL

### DISCUSSION THREAD

**Discussion by anstapol** [vercel/community#201] [Help] [answered]
Hey guys,

I'm using next-auth with next.js and [according to the documentation](https://next-auth.js.org/warnings#nextauth_url) I don't have to specify a `NEXTAUTH_URL` because it takes into an account the `VERCEL_URL` which in the [docs says](https://vercel.com/docs/concepts/projects/environment-variables) says that should point to the deployment URL.

So I was wondering, shouldn't it point to the main domain on production environments?

**Some more details just in case**
What happens on the frontend is that I land on the main domain login page:
`https://mydomain.com/auth/login`

But when I introduce my credentials it redirects me to
`https://app-[deploy-hash]-domain.vercel.app`

And obviously I'm not authenticated because the Cookie doesn't work for this domain.
And even if it would it is ugly & not right when the URL changes.

**Comment by Pieparker** [comment, marked as answer]
You're right! Nobody wants to see the domain to-and-fro. `VERCEL_URL` will always point to the specific deployment URL e.g. `[name-and-UID].vercel.app`. For custom domains, you'll want to configure this within the `NEXTAUTH_URL` var.

**Reply by anstapol** [reply]
Already did that :)

It's next auth "problem" then, thank you for your answer!
